### PR TITLE
Merge dynamic dimension values

### PR DESCRIPTION
**Pull request by philippjfr**
Merges the values from multiple composed DynamicMaps addressing https://github.com/ioam/holoviews/issues/920.

### CODE REVIEW

File: holoviews/core/util.py [hunk 1]
```diff
@@ -608,6 +608,23 @@ def python2sort(x,key=None):
     return itertools.chain.from_iterable(sorted(group, key=key) for group in groups)
 
 
+def merge_dimensions(dimensions_list):
+    """
+    Merges lists of fully or partially overlapping dimensions by
+    combining their values.
+    """
```
Comment (jlstevens) [inline holoviews/core/util.py]: The code is clear enough, but it might be nice to add a doctest example here (if we are using doctests).

**Comment by jlstevens** [comment]
Looks like my comment in the issue regarding a ``KeyError`` was confused. Anyway, merging dimensions with ``merge_dimensions`` makes sense - my only question, is there anywhere else we should be using it?

**Comment by philippjfr** [comment]
>my only question, is there anywhere else we should be using it?

I think this uses it everywhere it matters, i.e. when merging ``HoloMaps/DynamicMaps`` in an Overlay and when computing the dimensions during plotting.

**Comment by jlstevens** [comment]
Thanks for adding the doctest!

Happy to merge once the tests pass.

**Comment by philippjfr** [comment]
Some syntax error due to the test being multi-line.

**Comment by philippjfr** [comment]
Ready to merge when it passes.

**Comment by jlstevens** [comment]
Tests have passed. Merging.

**Pull request merged**